site/gdocs/ArticleBlock.tsx

```diff
@@ -272,45 +275,55 @@ export default function ArticleBlock({
                                     "h1-semibold",
                                     getLayout("heading", containerType),
                                     {
-                                        "has-supertitle": supertitleText,
+                                        "has-supertitle": supertitle
+                                            ? spansToUnformattedPlainText(
+                                                  supertitle
+                                              )
+                                            : "",
                                     }
                                 )}
                                 id={id}
                             >
-                                {supertitleText ? (
+                                {supertitle ? (
                                     <div className="article-block__heading-supertitle overline-black-caps">
-                                        {supertitleText}
+                                        {renderSpans(supertitle)}
                                     </div>
                                 ) : null}
-                                {text}
+                                {renderSpans(text)}
                             </h2>
                         </>
                     )
                 })
                 .with({ type: "heading", level: 3 }, (block) => {
-                    const {
-                        supertitle,
-                        text: { text },
-                    } = block
-                    const supertitleText = supertitle?.text || ""
-                    const id = urlSlug(`${supertitleText}-${text}`)
+                    const { supertitle, text } = block
+                    const id = supertitle
```

Comment:
Oops, I wrote out a different version of this thinking that this was causing the `.exhaustive()` failure only to re-read the code and see that the ternary was making that impossible.

Oh well, for your reference, just in case 🙂 

```js
const unformattedPlainText = [ supertitle, text ].filter().map(spansToUnformattedPlainText).join('-')
const slug = urlSlug(unformattedPlainText)

```